core/src/main/java/org/apache/iceberg/SnapshotProducer.java

```diff
@@ -89,6 +89,8 @@ public void accept(String file) {
 
   private ExecutorService workerPool = ThreadPools.getWorkerPool();
 
+  protected String toBranch = null;
```

Comment:
Imo, I think the name targetBranch is more clear.  it's used in a few other places as well. Also if we're assigning it, should we just assign it to main instead of null?